e2e/utils/pages/homepage.page.ts

```diff
@@ -0,0 +1,84 @@
+import {
+	HERO_ANIMATION_CANVAS,
+	LOGIN_BUTTON,
+	TOKENS_SKELETONS_INITIALIZED
+} from '$lib/constants/test-ids.constant';
+import { type InternetIdentityPage } from '@dfinity/internet-identity-playwright';
+import { type Page } from '@playwright/test';
+import { HOMEPAGE_URL, LOCAL_REPLICA_URL } from '../constants/e2e.constants';
+
+type HomepageParams = {
+	page: Page;
+};
+
+type HomepageLoggedInParams = {
+	iiPage: InternetIdentityPage;
+} & HomepageParams;
+
+abstract class Homepage {
+	readonly #page: Page;
+
+	protected constructor(page: Page) {
+		this.#page = page;
+	}
+
+	private async hideHeroAnimation(): Promise<void> {
+		await this.#page
+			.getByTestId(HERO_ANIMATION_CANVAS)
+			.evaluate((element) => (element.style.display = 'none'));
+	}
+
+	private async goto(): Promise<void> {
+		await this.#page.goto(HOMEPAGE_URL);
+	}
+
+	protected async waitForHomepageReady(): Promise<void> {
+		await this.goto();
+		await this.#page.getByTestId(LOGIN_BUTTON).waitFor();
+		await this.hideHeroAnimation();
+	}
+
+	protected async waitForTokenSkeletonsInitialization(): Promise<void> {
+		await this.#page.getByTestId(TOKENS_SKELETONS_INITIALIZED).waitFor();
+	}
+
+	abstract waitForReady(): Promise<void>;
+}
+
+export class HomepageLoggedOut extends Homepage {
+	constructor({ page }: HomepageParams) {
+		super(page);
+	}
+
+	async waitForReady(): Promise<void> {
```

Comment:
Just sharing in case you like it: you can mark those functions as being the override of the abstract. Don't know if that's the standard though. Really just sharing the idea.

```
/**
  * @override
*/
async waitForReady(): Promise<void> {
  await this.waitForHomepageReady();
}
```